lib/rgb_matrix/engine.ex

```diff
@@ -110,7 +110,7 @@ defmodule RGBMatrix.Engine do
   defp cancel_timer(%{timer: nil} = state), do: state
 
   defp cancel_timer(state) do
-    Process.cancel_timer(state.timer)
+    _ = Process.cancel_timer(state.timer)
```

Comment:
I'd rather this be removed (technically it should be `:ok = Process.sleep(...)`, because that's the only valid return value, but I think the only reason this was added was because of dialyzer's `:unmatched_returns` option.  I'd rather us just remove that option than having to do this here.

Actually, come to think of it, I think we should just dialyzer run with its default options.  Let's remove `flags:` from the dialyzer opts.